Sources/Plasma/CoreLib/hsWindows.h

```diff
@@ -142,6 +143,55 @@ You can contact Cyan Worlds, Inc. by email legal@cyan.com
             ST::format_type(format, output, hr.fResult);
         }
     }
+
+    template<class, class...>
+    class plOptionalWinCall;
```

Comment:
@Hoikas For what it's worth, I wanted to play around with something similar for some Linux stuff, and ended up slightly modifying this to work with `dlopen`/`dlsym` too. I also tried to add a map of open libraries so we can avoid re-opening the same library multiple times if we're pulling a bunch of functions from it.
My code (which is probably not perfect): https://gist.github.com/dpogue/6ac36a0fdd78397211441825b4ad5405

Reply:
> I also tried to add a map of open libraries so we can avoid re-opening the same library multiple times if we're pulling a bunch of functions from it.

My thought process with `plOptionalWincall` was that there would be a fairly centralized place for the function pointers to only be bound once, like all of the DPI stuff only ever being inside of the `plWinDpi` singleton. I think it would make more sense to just bind any other function pointers we need in a single TU at application start up time.

Reply:
yeah, but even within plWinDpi you're pulling 9 functions from user32, which is going to cause `LoadLibraryW`/`FreeLibrary` to get called 9 times for the same library. That's what I was hoping to avoid by caching the loaded library.

Reply:
That's true, but `USER32` is already loaded, so we're really just looking up the library by string name, incrementing the reference counter, then decrementing the reference counter. I'm not sure we really need to micro-optimize that.

Reply:
I would still sorta prefer to see this made not Windows-specific (so I can use it for X11 vs Wayland soft linking) but I'm also fine dealing with that in a future PR.

My proposal (which supports variables as well as functions, but is missing the WINAPI calling convention stuff): https://gist.github.com/dpogue/6ac36a0fdd78397211441825b4ad5405